File: SFSimulator.Web/src/app/models/dungeon.ts
export type Dungeon = {
  name: string,
  position: number,
  type: DungeonType,
  enemies: DungeonEnemy[] 
}

export type DungeonEnemy = {
  name: string,
  position: number,
  level: number,
  isDefeated: boolean,
  isUnlocked: boolean,
  class: number
}
export enum DungeonType
{
    LightWorld = -1,
    Twister = 1,
    Tower = 2,
    ShadowWorld = 4
}

File: SFSimulator.Core/Fights/Dungeon/Dungeon.cs
namespace SFSimulator.Core
{
    public class Dungeon
    {
        public required List<DungeonEnemy> DungeonEnemies { get; set; }
        public required string Name { get; set; }
        public required int Position { get; set; }
        public bool IsUnlocked { get; set; } = false;
        public bool IsDefeated { get; set; } = false;
        public DungeonTypeEnum Type { get; set; } = DungeonTypeEnum.Default;
        public Func<DungeonUnlockContext, bool> UnlockResolve { get; set; } = (_) => true;
    }
}